src/background/host.ts

```diff
@@ -1,26 +1,87 @@
+import Debug from "debug";
 import Peer from "peerjs";
-import { Message, MessageType } from "../Message";
+import { Friend, LocalOutMessage, LocalPollMessage, MessageType, RemoteMessage } from "../Message";
+
+export interface FriendConnected extends Friend {
+    conn: Peer.DataConnection;
+}
 
 export class Host {
+    #log = Debug("peer:disconnected");
     #port: chrome.runtime.Port;
     #peer = new Peer();
-    #friends = new Set<Peer.DataConnection>();
 
-    public isReady = false;
-    public id: string | null = null;
+    #friends = new Map<string, FriendConnected>();
+
+    public personalData: Friend | null = null;
     // eslint-disable-next-line @typescript-eslint/no-empty-function
     public onChangeFriends: (count: number, delta: number) => void = () => {};
 
-    #handleConnect = (conn: Peer.DataConnection): void => {
-        this.#friends.add(conn);
-        this.onChangeFriends(this.#friends.size, 1);
+    #ensureConnections = (friends: Friend[], conn: Peer.DataConnection): void => {
+        if (!this.personalData) {
+            return;
+        }
+
+        for (const member of friends) {
```

Comment:
do we care about users that are in our local map but weren't part of the users in the message we received

Reply:
uhm. maybe (?)

in my testing, if a user disconnects from one, they disconnect from all at pretty much the exact same time. the only case where this wouldn't happen is if an error occurs for one of their `FriendConnected` peers but not the rest.

right now we don't handle errors really at all .... and i think that's fine for now. down the road, we could be smart about `conn.on('error'` and knowing when/if to do`this.#friend.delete()` on that peer. Then anyone emitting an event would just signal to remake that connection for the two that had an error, while everyone else just keeps on chirping.